controllers/watch_test.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"testing"
+
+	"reflect"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	clusterv1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func TestMachineToInfrastructureMapFunc(t *testing.T) {
+	var testcases = []struct {
+		name    string
+		input   schema.GroupVersionKind
+		request *corev1.ObjectReference
+		output  []reconcile.Request
+	}{
+		{
+			name: "reconcile infra-1",
+			input: schema.GroupVersionKind{
+				Group:   "foo.cluster.sigs.k8s.io",
```

Comment:
Today, with the guidance from SIG Architecture, we decided to use `x-k8s.io` instead of `sigs.k8s.io`. Please make this change anywhere you have `sigs.k8s.io` in this PR. If there are other places in the code that use the old name, we can open a new PR to fix them. Thanks.